# Final project

## Before starting...

In [1]:
# package import
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
sns.set()

In [2]:
# load datasets
df_athletes = pd.read_csv('data/athletes.csv')
df_athletes.head()

ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season       City          Sport  \
0  CHN  1992 Summer  1992  Summer  Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer     London           Judo   
2  DEN  1920 Summer  1920  Summer  Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer      Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter    Calgary  Speed Skating   

                              Event Medal  
0       Basketball Men's Basketball   NaN  
1      Judo Men's Extra-Lightweight   NaN  
2           Football Men's Football   NaN  
3       Tug-Of-War Men's Tug-Of-War  Gold  
4  Speed Skating Women's 500 metres   NaN

In [3]:
df_athletes = df_athletes.drop( columns=["Name","ID","Games","Event"])
df_athletes=df_athletes.fillna(0)
df_athletes.head(20)

Sex   Age  Height  Weight            Team  NOC  Year  Season         City  \
0    M  24.0   180.0    80.0           China  CHN  1992  Summer    Barcelona   
1    M  23.0   170.0    60.0           China  CHN  2012  Summer       London   
2    M  24.0     0.0     0.0         Denmark  DEN  1920  Summer    Antwerpen   
3    M  34.0     0.0     0.0  Denmark/Sweden  DEN  1900  Summer        Paris   
4    F  21.0   185.0    82.0     Netherlands  NED  1988  Winter      Calgary   
5    F  21.0   185.0    82.0     Netherlands  NED  1988  Winter      Calgary   
6    F  25.0   185.0    82.0     Netherlands  NED  1992  Winter  Albertville   
7    F  25.0   185.0    82.0     Netherlands  NED  1992  Winter  Albertville   
8    F  27.0   185.0    82.0     Netherlands  NED  1994  Winter  Lillehammer   
9    F  27.0   185.0    82.0     Netherlands  NED  1994  Winter  Lillehammer   
10   M  31.0   188.0    75.0   United States  USA  1992  Winter  Albertville   
11   M  31.0   188.0    75.0   United States  USA  1992  Winter  Albertville   
12   M  31.0   188.0    75.0   United States  USA  1992  Winter  Albertville   
13   M  31.0   188.0    75.0   United States  USA  1992  Winter  Albertville   
14   M  33.0   188.0    75.0   United States  USA  1994  Winter  Lillehammer   
15   M  33.0   188.0    75.0   United States  USA  1994  Winter  Lillehammer   
16   M  33.0   188.0    75.0   United States  USA  1994  Winter  Lillehammer   
17   M  33.0   188.0    75.0   United States  USA  1994  Winter  Lillehammer   
18   M  31.0   183.0    72.0   United States  USA  1992  Winter  Albertville   
19   M  31.0   183.0    72.0   United States  USA  1992  Winter  Albertville   

                   Sport Medal  
0             Basketball     0  
1                   Judo     0  
2               Football     0  
3             Tug-Of-War  Gold  
4          Speed Skating     0  
5          Speed Skating     0  
6          Speed Skating     0  
7          Speed Skating     0  
8          Speed Skating     0  
9          Speed Skating     0  
10  Cross Country Skiing     0  
11  Cross Country Skiing     0  
12  Cross Country Skiing     0  
13  Cross Country Skiing     0  
14  Cross Country Skiing     0  
15  Cross Country Skiing     0  
16  Cross Country Skiing     0  
17  Cross Country Skiing     0  
18  Cross Country Skiing     0  
19  Cross Country Skiing     0

In [4]:
df_athletes=df_athletes.sort_values(by='Team')
df_athletes.head()

Sex   Age  Height  Weight                   Team  NOC  Year  Season  \
67840    M  23.0   167.0    70.0            30. Februar  AUT  1952  Summer   
254456   M  44.0   176.0     0.0            30. Februar  AUT  1952  Summer   
64703    M  38.0     0.0     0.0  A North American Team  MEX  1900  Summer   
263255   M   0.0     0.0     0.0  A North American Team  USA  1900  Summer   
64706    M  44.0     0.0     0.0  A North American Team  MEX  1900  Summer   

            City    Sport   Medal  
67840   Helsinki  Sailing       0  
254456  Helsinki  Sailing       0  
64703      Paris     Polo  Bronze  
263255     Paris     Polo  Bronze  
64706      Paris     Polo  Bronze

In [5]:
df_athletes.isna().sum()

Sex       0
Age       0
Height    0
Weight    0
Team      0
NOC       0
Year      0
Season    0
City      0
Sport     0
Medal     0
dtype: int64

There are no longer Na values, we can continue

In [6]:
df_athletes.drop( df_athletes[ df_athletes['Year'] < 1960].index , inplace=True)

In [7]:
df_athletes.head()

Sex   Age  Height  Weight         Team  NOC  Year  Season    City  \
73071    M  46.0   173.0    73.0    Acipactli  MEX  1964  Summer   Tokyo   
193165   M  52.0   180.0    84.0    Acipactli  MEX  1964  Summer   Tokyo   
51134    M  44.0   171.0    69.0    Acipactli  MEX  1964  Summer   Tokyo   
78903    M  28.0   178.0    57.0  Afghanistan  AFG  1980  Summer  Moskva   
267283   M  20.0   168.0    58.0  Afghanistan  AFG  1960  Summer    Roma   

            Sport Medal  
73071     Sailing     0  
193165    Sailing     0  
51134     Sailing     0  
78903      Boxing     0  
267283  Athletics     0

In [8]:
df_athletes.isna().sum()

Sex       0
Age       0
Height    0
Weight    0
Team      0
NOC       0
Year      0
Season    0
City      0
Sport     0
Medal     0
dtype: int64

There are no longer Na values, we can continue

In [9]:
df_athletes.loc[df_athletes.NOC == 'NED', 'NOC'] = 'NER'

In [10]:
#df_summer = df_athletes[df_athletes['Season']=='Summer']
#df_summer.head()

In [11]:
#df_winter = df_athletes[df_athletes['Season']=='Winter']
#df_winter.head()

In [12]:
df_countries = pd.read_csv('data/countries.csv')
df_countries.head()

Country                               Region  Population  \
0     Afghanistan         ASIA (EX. NEAR EAST)             31056997   
1         Albania   EASTERN EUROPE                          3581655   
2         Algeria   NORTHERN AFRICA                        32930091   
3  American Samoa   OCEANIA                                   57794   
4         Andorra   WESTERN EUROPE                            71201   

   Area (sq. mi.) Pop. Density (per sq. mi.) Coastline (coast/area ratio)  \
0          647500                       48,0                         0,00   
1           28748                      124,6                         1,26   
2         2381740                       13,8                         0,04   
3             199                      290,4                        58,29   
4             468                      152,1                         0,00   

  Net migration Infant mortality (per 1000 births)  GDP ($ per capita)  \
0         23,06                             163,07               700.0   
1         -4,93                              21,52              4500.0   
2         -0,39                                 31              6000.0   
3        -20,71                               9,27              8000.0   
4           6,6                               4,05             19000.0   

  Literacy (%) Phones (per 1000) Arable (%) Crops (%) Other (%) Climate  \
0         36,0               3,2      12,13      0,22     87,65       1   
1         86,5              71,2      21,09      4,42     74,49       3   
2         70,0              78,1       3,22      0,25     96,53       1   
3         97,0             259,5         10        15        75       2   
4        100,0             497,2       2,22         0     97,78       3   

  Birthrate Deathrate Agriculture Industry Service  
0      46,6     20,34        0,38     0,24    0,38  
1     15,11      5,22       0,232    0,188   0,579  
2     17,14      4,61       0,101      0,6   0,298  
3     22,46      3,27         NaN      NaN     NaN  
4      8,71      6,25         NaN      NaN     NaN

In [13]:
display(df_countries.isna().sum(),df_countries.count())

Country                                0
Region                                 0
Population                             0
Area (sq. mi.)                         0
Pop. Density (per sq. mi.)             0
Coastline (coast/area ratio)           0
Net migration                          3
Infant mortality (per 1000 births)     3
GDP ($ per capita)                     1
Literacy (%)                          18
Phones (per 1000)                      4
Arable (%)                             2
Crops (%)                              2
Other (%)                              2
Climate                               22
Birthrate                              3
Deathrate                              4
Agriculture                           15
Industry                              16
Service                               15
dtype: int64

Country                               227
Region                                227
Population                            227
Area (sq. mi.)                        227
Pop. Density (per sq. mi.)            227
Coastline (coast/area ratio)          227
Net migration                         224
Infant mortality (per 1000 births)    224
GDP ($ per capita)                    226
Literacy (%)                          209
Phones (per 1000)                     223
Arable (%)                            225
Crops (%)                             225
Other (%)                             225
Climate                               205
Birthrate                             224
Deathrate                             223
Agriculture                           212
Industry                              211
Service                               212
dtype: int64

22<<<227 so we might just drop those Na values

In [14]:
df_countries=df_countries.dropna()

In [15]:
df_countries.isna().sum().sum()

0

No Na values in the dataset. We can continue

In [16]:
df_countries.head()

Country                               Region  Population  \
0        Afghanistan         ASIA (EX. NEAR EAST)             31056997   
1            Albania   EASTERN EUROPE                          3581655   
2            Algeria   NORTHERN AFRICA                        32930091   
6           Anguilla               LATIN AMER. & CARIB           13477   
7  Antigua & Barbuda               LATIN AMER. & CARIB           69108   

   Area (sq. mi.) Pop. Density (per sq. mi.) Coastline (coast/area ratio)  \
0          647500                       48,0                         0,00   
1           28748                      124,6                         1,26   
2         2381740                       13,8                         0,04   
6             102                      132,1                        59,80   
7             443                      156,0                        34,54   

  Net migration Infant mortality (per 1000 births)  GDP ($ per capita)  \
0         23,06                             163,07               700.0   
1         -4,93                              21,52              4500.0   
2         -0,39                                 31              6000.0   
6         10,76                              21,03              8600.0   
7         -6,15                              19,46             11000.0   

  Literacy (%) Phones (per 1000) Arable (%) Crops (%) Other (%) Climate  \
0         36,0               3,2      12,13      0,22     87,65       1   
1         86,5              71,2      21,09      4,42     74,49       3   
2         70,0              78,1       3,22      0,25     96,53       1   
6         95,0             460,0          0         0       100       2   
7         89,0             549,9      18,18      4,55     77,27       2   

  Birthrate Deathrate Agriculture Industry Service  
0      46,6     20,34        0,38     0,24    0,38  
1     15,11      5,22       0,232    0,188   0,579  
2     17,14      4,61       0,101      0,6   0,298  
6     14,17      5,34        0,04     0,18    0,78  
7     16,93      5,37       0,038     0,22   0,743

In [17]:
df_GDP = pd.read_csv('data/GDP.csv')
df_GDP.head()

Country Name Country Code                Indicator Name  Indicator Code  \
0        Aruba          ABW  GDP per capita (current US$)  NY.GDP.PCAP.CD   
1  Afghanistan          AFG  GDP per capita (current US$)  NY.GDP.PCAP.CD   
2       Angola          AGO  GDP per capita (current US$)  NY.GDP.PCAP.CD   
3      Albania          ALB  GDP per capita (current US$)  NY.GDP.PCAP.CD   
4      Andorra          AND  GDP per capita (current US$)  NY.GDP.PCAP.CD   

        1960       1961       1962       1963       1964        1965  ...  \
0        NaN        NaN        NaN        NaN        NaN         NaN  ...   
1  59.773194  59.860874  58.458015  78.706388  82.095231  101.108305  ...   
2        NaN        NaN        NaN        NaN        NaN         NaN  ...   
3        NaN        NaN        NaN        NaN        NaN         NaN  ...   
4        NaN        NaN        NaN        NaN        NaN         NaN  ...   

           2010          2011          2012          2013          2014  \
0  23512.602596  24985.993281  24713.698045  25025.099563  25533.569780   
1    543.303042    591.162346    641.872034    637.165044    613.856333   
2   3587.883798   4615.468028   5100.095808   5254.882338   5408.410496   
3   4094.358816   4437.177794   4247.614342   4413.082887   4578.667934   
4  39736.354063  41100.729938  38392.943901  40626.751632  42300.334128   

           2015          2016          2017          2018  2019  
0  25796.380251  25239.600411  25630.266492           NaN   NaN  
1    578.466353    547.228110    556.302139    520.896603   NaN  
2   4166.979684   3506.072885   4095.812942   3432.385736   NaN  
3   3952.830781   4124.108543   4532.889198   5253.630064   NaN  
4  36039.653496  37224.108916  39134.393371  42029.762737   NaN  

[5 rows x 64 columns]

In [18]:
df_GDP=df_GDP.drop( columns="Country Name")

In [19]:
df_GDP.head()

Country Code                Indicator Name  Indicator Code       1960  \
0          ABW  GDP per capita (current US$)  NY.GDP.PCAP.CD        NaN   
1          AFG  GDP per capita (current US$)  NY.GDP.PCAP.CD  59.773194   
2          AGO  GDP per capita (current US$)  NY.GDP.PCAP.CD        NaN   
3          ALB  GDP per capita (current US$)  NY.GDP.PCAP.CD        NaN   
4          AND  GDP per capita (current US$)  NY.GDP.PCAP.CD        NaN   

        1961       1962       1963       1964        1965        1966  ...  \
0        NaN        NaN        NaN        NaN         NaN         NaN  ...   
1  59.860874  58.458015  78.706388  82.095231  101.108305  137.594352  ...   
2        NaN        NaN        NaN        NaN         NaN         NaN  ...   
3        NaN        NaN        NaN        NaN         NaN         NaN  ...   
4        NaN        NaN        NaN        NaN         NaN         NaN  ...   

           2010          2011          2012          2013          2014  \
0  23512.602596  24985.993281  24713.698045  25025.099563  25533.569780   
1    543.303042    591.162346    641.872034    637.165044    613.856333   
2   3587.883798   4615.468028   5100.095808   5254.882338   5408.410496   
3   4094.358816   4437.177794   4247.614342   4413.082887   4578.667934   
4  39736.354063  41100.729938  38392.943901  40626.751632  42300.334128   

           2015          2016          2017          2018  2019  
0  25796.380251  25239.600411  25630.266492           NaN   NaN  
1    578.466353    547.228110    556.302139    520.896603   NaN  
2   4166.979684   3506.072885   4095.812942   3432.385736   NaN  
3   3952.830781   4124.108543   4532.889198   5253.630064   NaN  
4  36039.653496  37224.108916  39134.393371  42029.762737   NaN  

[5 rows x 63 columns]

In [20]:
df_GDP.describe()

1960         1961         1962         1963         1964  \
count   133.000000   134.000000   136.000000   136.000000   136.000000   
mean    477.090548   510.997180   537.759283   570.571590   620.118759   
std     597.614845   642.387548   674.781661   714.643153   782.415979   
min      40.537115    40.689390    34.790581    40.752237    41.083814   
25%     107.949993   109.211954   114.955692   123.204376   124.980086   
50%     231.025649   217.409205   232.156045   252.392332   259.622516   
75%     489.351689   561.473554   634.966539   666.844461   752.171293   
max    3007.123445  3066.562869  3243.843078  3374.515171  3573.941185   

              1965         1966         1967         1968         1969  ...  \
count   146.000000   147.000000   150.000000   154.000000   154.000000  ...   
mean    687.168945   730.812250   747.106490   766.201365   828.090061  ...   
std     874.030595   940.758849   971.350324  1002.331545  1076.088550  ...   
min      45.989354    37.488783    46.642770    48.784045    51.794609  ...   
25%     141.029619   145.767969   157.983788   159.262062   161.937705  ...   
50%     273.655226   277.233872   278.460057   291.141585   305.373736  ...   
75%     854.120036   882.906899   875.547972   805.809962   860.595621  ...   
max    4443.405272  4571.181955  4336.426587  4695.923390  5032.144743  ...   

                2010           2011           2012           2013  \
count     251.000000     252.000000     251.000000     252.000000   
mean    14423.626372   15903.921344   15822.664863   16354.589444   
std     21710.830053   23961.297976   23150.370578   24726.589825   
min       234.235647     249.577979     252.358980     256.976003   
25%      1575.699381    1738.219933    1873.610010    1957.578334   
50%      5324.617040    5968.327288    6528.972218    6792.886617   
75%     16328.315060   18398.905381   18291.925068   19283.183735   
max    150585.448911  168785.940809  157515.899069  177593.351895   

                2014           2015           2016           2017  \
count     251.000000     250.000000     250.000000     246.000000   
mean    16356.195139   14707.131596   14747.116816   14515.124497   
std     25092.278876   22369.580670   22472.843662   20608.291936   
min       248.844951     293.465423     282.149682     293.009062   
25%      2035.213733    2009.418262    2050.138368    2006.922358   
50%      6600.056650    6036.798975    5813.802359    6170.481787   
75%     18999.005287   16666.300061   16998.588687   16409.835459   
max    189170.895671  167290.939984  169904.182096  166726.051926   

                2018  2019  
count     231.000000   0.0  
mean    14148.865580   NaN  
std     19413.684253   NaN  
min       275.429603   NaN  
25%      2021.886228   NaN  
50%      6202.162267   NaN  
75%     16426.431107   NaN  
max    114340.498742   NaN  

[8 rows x 60 columns]

## Merge first two datatsets

In [21]:
df_athletes = df_athletes.rename(columns={'Team': 'Country'})
df_athletes=df_athletes.set_index(df_athletes["Country"]).transpose()
df_countries=df_countries.set_index(df_countries["Country"]).transpose()
display(df_athletes.head(),df_countries.head())


Country  Acipactli  Acipactli  Acipactli  Afghanistan  Afghanistan  \
Sex              M          M          M            M            M   
Age             46         52         44           28           20   
Height         173        180        171          178          168   
Weight          73         84         69           57           58   
Country  Acipactli  Acipactli  Acipactli  Afghanistan  Afghanistan   

Country  Afghanistan  Afghanistan  Afghanistan  Afghanistan  Afghanistan  ...  \
Sex                M            M            M            F            M  ...   
Age               20           20           22           18           21  ...   
Height           168            0          172          180          164  ...   
Weight            58            0           70           56           51  ...   
Country  Afghanistan  Afghanistan  Afghanistan  Afghanistan  Afghanistan  ...   

Country  Zimbabwe  Zimbabwe  Zimbabwe  Zimbabwe  Zimbabwe  Zimbabwe  Zimbabwe  \
Sex             F         M         M         M         F         M         M   
Age            27        35        30        27        26        24        24   
Height        155       172       182       196       172       163       163   
Weight         45        63        80        89        56        55        55   
Country  Zimbabwe  Zimbabwe  Zimbabwe  Zimbabwe  Zimbabwe  Zimbabwe  Zimbabwe   

Country  Zimbabwe  Zimbabwe  Zimbabwe  
Sex             M         M         M  
Age            26        20        42  
Height        163       163       182  
Weight         53        55        80  
Country  Zimbabwe  Zimbabwe  Zimbabwe  

[5 rows x 209072 columns]

Country                                      Afghanistan   \
Country                                      Afghanistan    
Region                      ASIA (EX. NEAR EAST)            
Population                                       31056997   
Area (sq. mi.)                                     647500   
Pop. Density (per sq. mi.)                           48,0   

Country                                                Albania   \
Country                                                Albania    
Region                      EASTERN EUROPE                        
Population                                              3581655   
Area (sq. mi.)                                            28748   
Pop. Density (per sq. mi.)                                124,6   

Country                                                Algeria   \
Country                                                Algeria    
Region                      NORTHERN AFRICA                       
Population                                             32930091   
Area (sq. mi.)                                          2381740   
Pop. Density (per sq. mi.)                                 13,8   

Country                                   Anguilla        Antigua & Barbuda   \
Country                                   Anguilla        Antigua & Barbuda    
Region                      LATIN AMER. & CARIB      LATIN AMER. & CARIB       
Population                                    13477                    69108   
Area (sq. mi.)                                  102                      443   
Pop. Density (per sq. mi.)                    132,1                    156,0   

Country                                  Argentina               Armenia   \
Country                                  Argentina               Armenia    
Region                      LATIN AMER. & CARIB      C.W. OF IND. STATES    
Population                                 39921833               2976372   
Area (sq. mi.)                              2766890                 29800   
Pop. Density (per sq. mi.)                     14,4                  99,9   

Country                                      Aruba   \
Country                                      Aruba    
Region                      LATIN AMER. & CARIB       
Population                                    71891   
Area (sq. mi.)                                  193   
Pop. Density (per sq. mi.)                    372,5   

Country                                              Australia   \
Country                                              Australia    
Region                      OCEANIA                               
Population                                             20264082   
Area (sq. mi.)                                          7686850   
Pop. Density (per sq. mi.)                                  2,6   

Country                                                Austria   ...  \
Country                                                Austria   ...   
Region                      WESTERN EUROPE                       ...   
Population                                              8192880  ...   
Area (sq. mi.)                                            83870  ...   
Pop. Density (per sq. mi.)                                 97,7  ...   

Country                                         United Kingdom   \
Country                                         United Kingdom    
Region                      WESTERN EUROPE                        
Population                                             60609153   
Area (sq. mi.)                                           244820   
Pop. Density (per sq. mi.)                                247,6   

Country                                          United States   \
Country                                          United States    
Region                      NORTHERN AMERICA                      
Population                                            298444215   
Area (sq. mi.)                               

In [ ]:
#maybe put information about athlets from the same country under the same index

In [22]:
data1=pd.merge(df_athletes,df_countries)

MergeError: No common columns to perform merge on. Merge options: left_on=None, right_on=None, left_index=False, right_index=False

## Getting the min value of the last 4 years for the current year

In [ ]:
# fill the NaN values
#df_GDP = df_GDP.fillna(method='backfill', inplace = True, axis=1)

df_GDP = df_GDP.T
df_GDP.head()

In [ ]:
df_GDP = df_GDP.drop(['Indicator Name', 'Indicator Code', '2019', '2018'])

In [ ]:
df_GDP = df_GDP.fillna(method='backfill', axis = 0)
df_GDP = df_GDP.fillna(method='ffill', axis = 0)

In [ ]:
df_GDP = df_GDP.T

In [ ]:
df_GDP.head(5)

In [ ]:
df_GDP.describe()

In [ ]:
df_GDP_4years = df_GDP

In [ ]:
df_GDP_4years.set_index('Country Code', inplace=True)
df_GDP_4years.head()

In [ ]:
#def get_min_4years(x):
    #if(x[])
    #return x

In [ ]:
i = 1960
a = {}
while i < 2019:
    a[str(i)] = [str(i-1), str(i-2), str(i-3), str(i-4)]
    i+=1

In [ ]:
#df_GDP_4years[a.get('1964')].min(axis=1)
#df_GDP_4years[a[str('1964')]]

In [ ]:
df_GDP_new = df_GDP_4years.head()

In [ ]:
df_GDP_new

In [ ]:
#df_GDP_new[df_GDP_new.index == 'ABW']

In [ ]:
abw = df_GDP_new[a.get(str(1964))].min(axis=1)[df_GDP_new.index == 'ABW']
print(abw)
abw+20

In [ ]:
for i, country in df_GDP_new.iterrows(): 
    #print(i, year) 
    #print() 
    year = 2017
    #print("------------- ", i, "---------------")
    while(year > 1963):
        #print("-----------------")
        #print(country[str(year)], " = ", df_GDP_new[a.get(str(year))].min(axis=1)[df_GDP_new.index == str(i)])
        country[str(year)] = float(df_GDP_new[a.get(str(year))].min(axis=1)[df_GDP_new.index == str(i)])
        #print("-----------------")
        year -= 1

#idx = 2017
#for index, row in df_GDP_new.iterrows():
    #print("--------- new row---------")    
    #print("index: ", idx, "\n", df_GDP_new[a.get(str(idx))].min(axis=1))
    #idx += -1
    #print(df_GDP_new[a.get(str(idx))].min(axis=1))
            #df_GDP_new.index(row.index()) = row
            #row[str(idx)] = df_GDP_new[a.get(str(idx))].min(axis=1)

In [ ]:
df_GDP_new

In [ ]:
#print(df_GDP_4years.iloc[249])

In [ ]:
#df_at = df_athletes
#n = df_GDP_4years[df_GDP_4years['Country Code'] == 'ALB']['1960']
#n

In [ ]:
#gdp = []

In [ ]:
#df_GDP_4years['Country Code']
#i = 0
#for index, row in df_at.iterrows():
    #i+= 1
    ##gdp.append(df_GDP_4years[df_GDP_4years['Country Code'] == row['NOC']][row['Year']])

In [ ]:
#i

In [ ]:
#ex = df_at.head(20)
#ex

In [ ]:
#i = 0
#ex2 = []
#for index, e in ex.iterrows():
    # get country
    #country = df_GDP_4years[df_GDP_4years['Country Code'] == e['NOC']]
    #year = str(e['Year'])
    
    #print("--------------------\n", year, "\n", country[year])

In [ ]:
#i = 0
#ex2 = []
#for index, e in ex.iterrows():
    # get country
    #country_idx = df_GDP_4years[df_GDP_4years['Country Code'] == e['NOC']].index
    # get GDP of year 
    #print(str(e['Year']))
    #gdp = df_GDP_4years.iloc[country_idx][str(e['Year'])]
    #print("\n--------\n gdp:\n", gdp, "\n-------------")
    #gdp = 
    #print(e['NOC'])
    #print(e['Year'])
    #print(e['Year'])
    #year = e['Year']
    #print(df_GDP_4years.loc[1])
    #df_GDP_4years[str(e['Year'])]
    #print(df_GDP_4years[e['Year']])
    #ex2.append(df_GDP_4years[df_GDP_4years['Country Code'] == e['NOC']]['2018'])

In [ ]:
#df = pd.DataFrame(ex2) 
#df

In [ ]:
#df_at['GDP'] = df_GDP_4years.loc[df_GDP_4years['Country Code'] == df_at['NOC']]['1960']